unittest/joint-spline.cpp

```diff
@@ -0,0 +1,422 @@
+  translation << 0.000757726, -0.399, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.939693, 0, -0.34202, 0, 1, 0, 0.34202, 0, 0.939693;
+  translation << -0.001, -0.3976, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.984808, 0, -0.173648, 0, 1, 0, 0.173648, 0, 0.984808;
+  translation << -0.0031, -0.3966, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.987363, 0, 0.158478, 0, 1, 0, -0.158478, 0, 0.987363;
+  translation << -0.00513719, -0.395264, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.980591, 0, 0.196066, 0, 1, 0, -0.196066, 0, 0.980591;
+  translation << -0.005227, -0.395149, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.94362, 0, 0.33103, 0, 1, 0, -0.33103, 0, 0.94362;
+  translation << -0.005435, -0.394792, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.882249, 0, 0.470783, 0, 1, 0, -0.470783, 0, 0.882249;
+  translation << -0.005574, -0.394507, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << 0.0493163, 0, 0.998783, 0, 1, 0, -0.998783, 0, 0.0493163;
+  translation << -0.005435, -0.394812, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+
+  rotation << -0.500004, 0, 0.866023, 0, 1, 0, -0.866023, 0, -0.500004;
+  translation << -0.00525, -0.396, 0;
+  ctrlFrames.push_back(pinocchio::SE3(rotation, translation));
+}
+
+BOOST_AUTO_TEST_SUITE(JointSpline)
+
+/// @brief Test on the knot vector generation
+BOOST_AUTO_TEST_CASE(makeKnots)
+{
+  int degree = 3;
+
+  std::vector<SE3> ctrlFrames;
+  ctrlFrames.push_back(SE3::Identity());
+  ctrlFrames.push_back(SE3::Random());
+  ctrlFrames.push_back(SE3::Random());
+
+  BOOST_CHECK_THROW(JointModelSpline(ctrlFrames, degree), std::invalid_argument);
+
+  ctrlFrames.push_back(SE3::Random());
+
+  JointModelSpline jmodel(ctrlFrames, degree);
+
+  // Check size
+  BOOST_CHECK(jmodel.knots.size() == (degree + ctrlFrames.size() + 1));
+
+  // Check Values
+  Eigen::VectorXd knots_expected(degree + ctrlFrames.size() + 1);
+  knots_expected << 0., 0., 0., 0., 1., 1., 1., 1.;
+  BOOST_CHECK(jmodel.knots == knots_expected);
+}
+
+/// @brief Test to make sure the relative motions are correct
+BOOST_AUTO_TEST_CASE(relativeMotions)
+{
+  int degree = 3;
+
+  std::vector<SE3> ctrlFrames;
+  ctrlFrames.push_back(SE3::Identity());
+  ctrlFrames.push_back(SE3::Random());
+  ctrlFrames.push_back(SE3::Random());
+  ctrlFrames.push_back(SE3::Random());
+
+  JointModelSpline jmodel(ctrlFrames, degree);
+
+  // Check size
+  BOOST_CHECK(jmodel.relativeMotions.size() == (ctrlFrames.size() - 1));
+
+  // check values
+  for (size_t i = 0; i < ctrlFrames.size() - 1; i++)
+    BOOST_CHECK(jmodel.relativeMotions[i].isApprox(
+      pinocchio::log6(ctrlFrames[i].inverse() * ctrlFrames[i + 1])));
```

Comment:
This test duplicate the `computeRelativeMotions` code.

To avoid that and have a more robust test we can generate ctrlFrames from random `Motion` and check that `jmodel.relativeMotions` contains the generated random `Motion`